### PR TITLE
Nataliia romanchuk/homework6

### PR DESCRIPTION
**Pull request by Vivanatalya**

### CODE REVIEW

File: nataliia_romanchuk/homework6/min_max.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    return res
+
+
+def min_list(*args):
+    min_value = None
+    for number in args:
+        if number is not None:
+            if min_value is None or number < min_value:
+                min_value = number
+    return min_value
```
Comment (fkudinov-mn) [inline nataliia_romanchuk/homework6/min_max.py:16]: why you need this? if you have minnn

File: nataliia_romanchuk/homework6/min_max.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+def minnn(*args, key=None):
+    key = key or (lambda i: i)
+    res = args[0]
+    for i in args:
+        if key(i) < key(res):
+            res = i
+    return res
```
Comment (fkudinov-mn) [inline nataliia_romanchuk/homework6/min_max.py:7]: where is maxxx

File: nataliia_romanchuk/homework6/min_max.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+def min_list(*args):
+    min_value = None
+    for number in args:
+        if number is not None:
+            if min_value is None or number < min_value:
+                min_value = number
+    return min_value
+
+
+def max_list(*args):
```
Comment (fkudinov-mn) [inline nataliia_romanchuk/homework6/min_max.py:19]: i would expect 


def maxxx(*args, key=None):

File: nataliia_romanchuk/homework6/min_max.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    max_value = None
+    for number in args:
+        if number is not None:
+            if max_value is None or number > max_value:
+                max_value = number
+    return max_value
+
+
+def sorted_list(*args, reverse=False):
```
Comment (fkudinov-mn) [inline nataliia_romanchuk/homework6/min_max.py:28]: should also use key param to specify sorting

File: nataliia_romanchuk/homework6/min_max.py (hunk 2 above)
Comment (Vivanatalya) [reply nataliia_romanchuk/homework6/min_max.py:7]: added

File: nataliia_romanchuk/homework6/min_max.py (hunk 1 above)
Comment (Vivanatalya) [reply nataliia_romanchuk/homework6/min_max.py:25]: removed

File: nataliia_romanchuk/homework6/min_max.py (hunk 3 above)
Comment (Vivanatalya) [reply nataliia_romanchuk/homework6/min_max.py:28]: added

File: tests/nataliia_romanchuk/homework_6/test_unpack.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from nataliia_romanchuk.homework6.unpack import unpack_recursive, unpack_while_loop
+import pytest
+
+@pytest.mark.parametrize("some_item", [[123, ["234", None], [[1], [23], [[123], 123, ["sdf", True]]]]])
+def test_unpack_while_loop(some_item):
+    assert unpack_while_loop(some_item) == [123, "234", None, 1, 23, 123, 123, "sdf", True]
+
+@pytest.mark.parametrize("some_item", [[123, ["234", None], [[1], [23], [[123], 123, ["sdf", True]]]]])
+def test_unpack_recursive(some_item):
+    assert unpack_recursive(some_item) == [123, "234", None, 1, 23, 123, 123, "sdf", True]
```
Comment (fkudinov-mn) [inline tests/nataliia_romanchuk/homework_6/test_unpack.py:10]: no need in parametrize when only 1 case

you can dus such parametrize:

```
@pytest.mark.parametrize("func", [unpack_while_loop, unpack_recursive])
def test_unpack(func):
    some_item = [123, ["234", None], [[1], [23], [[123], 123, ["sdf", True]]]]
    assert func(some_item) == [123, "234", None, 1, 23, 123, 123, "sdf", True]

```

**Review by kudinov-fedor — DISMISSED** [review]

File: tests/nataliia_romanchuk/homework_6/test_unpack.py (hunk 5 above)
Comment (Vivanatalya) [reply tests/nataliia_romanchuk/homework_6/test_unpack.py]: fixed

**Pull request merged**